src/fauxtoshop.cpp

```diff
@@ -1,517 +1,487 @@
+        cout << "Invalid input! Please try again" << endl;
+    }
+}
+
+/*
+ * Function: validateLoctionInput
+ * Usage: Ensure user input for specifying a location on the canvas/ window is valid,
+ * otherwise reprompt the user to re-enter
+ * The user can enter the location in '(<row>, <col>)' format or hit enter
+ * to do this with a mouse click
+ * Params: row - where the sticker is place, col - where the sticker is place
+ * ------------------------------------------------------------------------------------------------
+ * Returns: nothing. Void function
+*/
+void validateLocationInput(int& row, int& col){
+    while (true){
+        string locationInput = getLine(
+                    "Enter sticker location as '(<row>, <col>)'or hit enter to use the mouse: ");
+        // Validate that the input starts and ends with "("")" and contains ","
+        if (locationInput.empty()){
+            cout << "Click on the canvas to place your sticker :-)" << endl;
+            getMouseClickLocation(row, col);
+            break;
+        } else if (startsWith(locationInput,"(")
+                && endsWith(locationInput,")")
+                && stringContains(locationInput, ",")
+                ){
+            // Remove the '('')' for delimitting
+            locationInput = stringReplace(locationInput, "(", "", 1);
+            locationInput = stringReplace(locationInput, ")", "", 1);
+            // Delimit the input string using "," and convert it into vector
+            Vector<string> inputVector = stringSplit(locationInput, ",");
+            // Validate that the vector has 2 elements and both elements can be converted into an
+            // integer
+            if (inputVector.size() == 2
+                    && stringIsInteger(inputVector[0])
+                    && stringIsInteger(inputVector[1])
+                    ){
+                // Convert both elements into integers
+                row = stringToInteger(inputVector[0]);
+                col = stringToInteger(inputVector[1]);
+                if (row > 0 && col > 0){
+                    break;
+                }
+            }
+        }
+        cout << "Invalid input! Please try again" << endl;
+    }
+}
+
+/*
+ * Function: calculateRBGColourDifference
+ * Usage: Calculates the difference between 2 colours for edge detection and green screen effects
+ * Function:
+ * a. Separate the pixels into their RGB components
+ * b. Take difference between each pair of RGB values (e.g. 2 reds, 2 blues, 2 greens)
+ * c. Take absolute value between for each difference (use nteger abs function in <math> standard)
+ * d. Take a max of the three differences (max function for 2 values at a time)
+ * scatter radius to provide colour for the pixel in new Grid. Select again, if the random selection
+ * is out of bounds of orignal Grid
+ * Hint: use int data type
+ * Extension: Alternate calcalatons
+ * Params: pixel1_row - integer, pixel1_col - integer, pixel2_row - integer, pixel2_col - integer
```

Comment:
Remember to update this to match the actual parameters in the function prototype. :)

Same for all of your other functions.